README.md

```diff
@@ -1,634 +1,1855 @@
+of assets a borrower can use as collateral for a loan.*
+- **Interest-Free, Non-Compounding, or Compounding Interest** - lenders and borrowers have full
+flexibility over what the interest for the loans will be. They can even decide on what the
+compounding frequency will be.
+- **Naturally Incentivized Regular Payments** - borrower's are incentivized to make regular payments
+on loans to avoid penalties agreed upon during negotiations. Because of this, lenders may be more
+willing to lend since there will be more regularity in when they will receive payments. This
+regularity can help them manage risk.
+- **Direct Loan Payments to Lenders** - borrowers make each loan payment directly to the address
+specified by the lender. This means lenders don't need to wait until the end of the loan to claim
+the proceeds from each payment.
+- **Tradable Bonds** - when a borrower accepts an offer from a lender, the lender recieves an NFT
+that reflects ownership of the loan. In the case of a default, whoever owns this NFT is able to
+claim the remaining collateral. Furthermore, the owner of this NFT is able to update the required
+payment address for the loan, even after it has started. This NFT is effectively the loan bond that
+can be freely traded on a secondary market.
+- **Over-Collateralized, Under-Collateralized, or Fully-Collateralized Loans** - lenders and
+borrowers can set the relative values of the collateral assets to anything they want. If they set
+the relative values above the market price, that is an over-collateralized loan; at the market price
+would be a fully-collateralized loan; and below the market price would be an under-collateralized
+loan. If lenders think the value of the collateral will change over the course of the loan's
+lifetime, they can set the relative values in such a way to minimize this risk.
+- **A Single DApp Address for each Borrower** - every borrower gets their own personal DApp address.
+All negotiations involving the borrower take place in this address and all active loans live in this
+address. This makes it very easy for front-ends to integrate Cardano-Loans as well as for borrowers
+to manage the stake/voting power for assets in the DApp.
+- **Efficient Off-chain Sorting of Loans** - by using beacon tokens, users can easily query the
+protocol for any information they may want. These beacon queries can even be combined to create more
+complicated queries. For example, lenders can see all current Asks and optionally filter them by
+loan asset and/or offered collateral.
+- **Democratic Upgradability** - users decide if and when to upgrade to a new version of the DApp.
+No one can decide this for them, or force them to move to a new version (eg, by freezing all
+functionality for the current version).
+- **Full Composability** - this protocol is natively composable with all other DeFi DApps. It can
+even be composed with itself: a borrower can accept an offer in the same transaction where they make
+a payment.
```

Comment:
Just a general comment about trade-offs with restriction-less financial instrument features:

Whilst in keeping with a permissionless defi system, this design choice impacts composability in the sense that it becomes more difficult to collect similar financial instruments into baskets.

But maybe you are right.  Maybe things will go this way because AI may make pricing such varied instruments easier and thus a large parcel composed of such instruments could be more easily valued???  TradFi markets re-package loans into these huge collateralied debt obligations so that they can trade fractionally for greater liquidity, and so that large institutional lenders can participate within their regulatory bounds.

Maybe that could be another project for you?  Developing software to accurately price bundles of these cardano-loans and re-package them into CDOs which are in-turn divided up into different tranches to meet the requirements of large lenders?

Reply:
The CDOs was one of the applications I had in mind with this design. I am not personally interested in creating software for them, but I do think there is value in someone doing it and I think there is enough financial incentive for it to eventually happen.

> Whilst in keeping with a permissionless defi system, this design choice impacts composability in the sense that it becomes more difficult to collect similar financial instruments into baskets.

I don't think this is true since the beacon tokens make filtering the entire DApp market very easy. For the CDO example, it is theoretically easy to cross-reference borrowers' credit histories with the target active loans; it would just require some moderately powerful hardware.